### PR TITLE
New page: namespace combinator

### PR DESCRIPTION
**Pull request by estelle**
While mentioned on the universal selector page and type selector pages, there was no documentation for this namespace separator, which is a type of CSS combinator.

### CODE REVIEW

File: files/en-us/web/css/namespace_separator/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [@namespace](/en-US/docs/Web/CSS/@namespace)
```
Comment (Josh-Cena) [inline files/en-us/web/css/namespace_separator/index.md:104]:
```suggestion
- [`@namespace`](/en-US/docs/Web/CSS/@namespace)
```

**Comment by Josh-Cena** [comment]
You would also need a link to this page on https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors

**Comment by estelle** [comment]
> You would also need a link to this page on https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors

This is part of a much larger project, https://github.com/openwebdocs/project/issues/164. CSS_selectors is being completely rewritten but I didn't want to make that PR too huge and this page was "ready"

**Comment by Josh-Cena** [comment]
Ah, sure—don't let me block it, then. It's still better to get a review from someone more familiar with CSS docs.

File: files/en-us/web/css/namespace_separator/index.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+The CSS declares two namespaces, then assigns styles to links globally, to links in no name space, any namespace or no namespace, and than to two different named namespaces.
+
+```css
+@namespace myNamespace url("http://www.w3.org/2000/svg");
+@namespace defaultNameSpace url("http://www.w3.org/1999/xhtml");
+/* all namespaces or no namespace */
+a {
+  font-size: 1.4rem;
+}
+/* no name space */
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:52]:
```suggestion
/* no namespace */
```

File: files/en-us/web/css/namespace_separator/index.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  fill: red;
+  font-style: italic;
+}
+/* only the myNamespace namespace */
+myNamespace|a {
+  color: green;
+  fill: green;
+}
+/* all HTML is in a namespace */
+defaultNameSpace|a {
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:69]: This example isn't different from `myNamespace` one. :confused:

File: files/en-us/web/css/namespace_separator/index.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+/* all namespaces or no namespace */
+a {
+  font-size: 1.4rem;
+}
+/* no name space */
+|a {
+  text-decoration: wavy overline lime;
+  font-weight: bold;
+}
+/* all namespaces or no namespace */
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:57]:
```suggestion
/* all namespaces (including no namespace) */
```

File: files/en-us/web/css/namespace_separator/index.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+## Syntax
+
+```css
+namespace|element { style properties }
+```
+
+## Examples
+
+#### CSS
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:41]:
```suggestion
### CSS
```
Jumping from level 2 to level 4 header. Don't know why the linter hasn't complained.

File: files/en-us/web/css/namespace_separator/index.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+{{CSSRef("Selectors")}}
+
+The **namespace separator** (`|`) separates the selector from the namespace, identifying the namespace, or lack thereof, for a type selector.
+
+```css
+/* Links in the namespace named myNameSpace */
+myNameSpace|a {
+  font-weight: bold;
+}
+/* paragraphs in any or no namespace */
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:17]:
```suggestion
/* paragraphs in any namespace (including no namespace) */
```

**Review by OnkarRuikar — COMMENTED** [review]
I know in the example you are trying to demo HTML and SVG namespaces. But it feels like the example is trying to demo the default namespace `@namespace url(...)` concept.

File: files/en-us/web/css/namespace_separator/index.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+## Examples
+
+#### CSS
+
+The CSS declares two namespaces, then assigns styles to links globally, to links in no name space, any namespace or no namespace, and than to two different named namespaces.
+
+```css
+@namespace myNamespace url("http://www.w3.org/2000/svg");
+@namespace defaultNameSpace url("http://www.w3.org/1999/xhtml");
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/namespace_separator/index.md:47]: That was the name I gave the namespace. I thought that made it clearer, but I guess it doesn't ;)

[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/namespace_separator/index.md]: I clarified the code example and the surrounding text

File: files/en-us/web/css/namespace_separator/index.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+/* all namespaces or no namespace */
```
[the remark this answers is not in the document]
Comment (estelle) [reply files/en-us/web/css/namespace_separator/index.md]: Thanks. This was super helfpul. I added a second example to address this, and changed the names away from using "default" to make it less confusing.

File: files/en-us/web/css/namespace_separator/index.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+const a = document.createElementNS("", "a");
+a.href = "#";
+a.appendChild(document.createTextNode("Link created without a namespace"));
+
+document.body.appendChild(a);
+```
+
+#### CSS
+
+We declare a namespace with {{cssxref("@namespace")}}. By ommitting the name for the namespace, the `@namespace` declaration creates a default namespace. name for the name.
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:135]:
```suggestion
We declare a namespace with {{cssxref("@namespace")}}. By omitting the name for the namespace, the `@namespace` declaration creates a default namespace.
```

File: files/en-us/web/css/namespace_separator/index.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+In this example, we use JavaScript to create an element without a namespace and append it to the document. We set the SVG namespace to be the default namespace by defining the unnamed namespace with `@namespace`.
+
+> **Note:** If a default, or unnamed, namespace is defined, universal and type selectors apply only to elements in that namespace.
+
+#### HTML
+
+No namespaces are explicitly declared in the HTML or within the SVG.
+
+```html
+<p>In the default namespace <a href="#">has a link</a>.</p>
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:111]: We are setting SVG as the default names space in CSS. Seeing this default in the output creates confusion.
```suggestion
<p><a href="#">A link</a> in the implied HTML namespace.</p>
```

File: files/en-us/web/css/namespace_separator/index.md [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+
+/* `<svg>` and `<p>` in the default (set to SVG) namespace */
+svg,
+p {
+  border: 5px solid gold;
+}
+
+/* links outside of any namespace */
+|a {
+  text-decoration: wavy overline purple;
```
Comment (OnkarRuikar) [inline files/en-us/web/css/namespace_separator/index.md:154]:
```suggestion
  text-decoration: wavy underline purple;
```
Overline overlaps with the gold border on SVG link above. Underline would make it more distinct.

**Review by OnkarRuikar — COMMENTED** [review]
This is great! Couldn't find any other documentation online covering no namespace example.

File: files/en-us/web/css/namespace_separator/index.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [`@namespace`](/en-US/docs/Web/CSS/@namespace)
+- [`Document.createElementNS()](/en-US/docs/Web/API/Document/createElementNS) method
```
Comment (Josh-Cena) [inline files/en-us/web/css/namespace_separator/index.md:185]:
```suggestion
- [`Document.createElementNS()`](/en-US/docs/Web/API/Document/createElementNS) method
```

**Review by OnkarRuikar — COMMENTED** [review]
@estelle Thank you! Looks good to me!

File: files/en-us/web/css/namespace_separator/index.md (hunk 12 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/namespace_separator/index.md]: On next line, `Element.namespaceURI` can also be wrapped in inline code block: ``` [`Element.namespaceURI`](...) ```

File: files/en-us/web/css/namespace_separator/index.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [`@namespace`](/en-US/docs/Web/CSS/@namespace)
+- [`Document.createElementNS()`](/en-US/docs/Web/API/Document/createElementNS) method
+- [Element.namespaceURI](/en-US/docs/Web/API/Element/namespaceURI) property
```
Comment (estelle) [inline files/en-us/web/css/namespace_separator/index.md:186]:
```suggestion
- [`Element.namespaceURI`](/en-US/docs/Web/API/Element/namespaceURI) property
```

**Review by dipikabh — APPROVED** [review]
Approving as requested 🙂

**Comment by dipikabh** [comment]
The checks are waiting on "Checking for ability to merge automatically…". I can merge after that goes through.

**Pull request merged**